### ISSUE TITLE
Task: validate CUSTOMERS file before use

### ISSUE THREAD

**Issue by alexellis** [openfaas/openfaas-cloud#391]
## Expected Behaviour

We should validate all entries in the CUSTOMERS file before use - to prevent anyone getting their functions overriden by accident. See issue #314 

## Possible Solution

Since this is relying on a human element of not merging a PR to the customers file with two potentially clashing usernames, we should implement a unit test and sdk method in the sdk of OpenFaaS Cloud that can validate a list of usernames.

I.e.

OK 

```
alexellis
stefan
ivana
```

Not OK 

```
alexellis
alexellis-suffix
stefan
ivana
```

Not OK 

```
alexellis
stefan
stefan-prodan
ivana
```

OK 

```
alexellis
stefan-prodan
ivana
```

**Comment by alexellis** [comment]
The PR should be raised here: https://github.com/openfaas/openfaas-cloud/tree/master/sdk

**Comment by alexellis** [comment]
No need to test end-to-end, just via unit tests.

https://blog.alexellis.io/golang-writing-unit-tests/

**Comment by rajatjindal** [comment]
how about

```
alexellis   
stefan-prodan  
stefan  
ivana
```

and

```
alexellis
alexe
stefan
ivana
```

**Comment by alexellis** [comment]
What about it?

**Comment by alexellis** [comment]
alexe won't clash because of the separator being `-`

**Comment by alexellis** [comment]
`alexe` => `alexe-my-function`
`alexellis` => `alexellis-my-function`

**Comment by martindekov** [comment]
Derek close: fixed in #393
